src/main/java/Duke.java

```diff
@@ -1,10 +1,145 @@
+import duke.exception.DukeException;
+import duke.exception.TaskType;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.Scanner;
+
 public class Duke {
+
+    private static Task[] tasks = new Task[100];
+    private static Scanner sc = new Scanner(System.in);
+
+    public static void listTasks() {
+        if (Task.totalTasks == 0) {
+            System.out.println("No tasks yet!");
+        } else {
+            for (int i = 0; i < Task.totalTasks; i++) {
+                System.out.printf("%d.%s\n", i+1, tasks[i].toString());
+            }
+        }
+    }
+
+    public static void markTasksAsDone(String input) {
+        String[] inputArray = input.split(" ");
+
+        //completedIndex holds the index of valid integer(s) in inputArray (indicating index in tasklist)
+        int completedIndex;
+        for (String word: inputArray) {
+            if (word.equals("done")) {
+                continue;
+            } else {
+                completedIndex = Integer.parseInt(word);
+                //ensure that the index given is valid
+                if (completedIndex > 0 && completedIndex <= Task.totalTasks){
+                    tasks[completedIndex - 1].markAsDone();
+                } else {
+                    System.out.printf("duke.task.Task %d does not exist! Enter 'list' to view tasklist :)\n", completedIndex);
+                }
+            }
+        }
+    }
+
+    public static String[] extractDetailsFromInput(String input, String keyword) {
+        String[] inputArray = new String[2];
+        String inputWithoutKeyword = input.split(keyword)[1];
```

Comment:
Same dangerous code: what if there are multiple appearances of the same keyword in the command?